pkg/ccl/backupccl/datadriven_test.go

```diff
@@ -187,26 +200,89 @@ func (d *datadrivenTestState) getSQLDB(t *testing.T, server string, user string)
 // commands. The test files are in testdata/backup-restore. The following
 // syntax is provided:
 //
-// - "new-server name=<name> [share-io-dir=<name>]"
-//   Create a new server with the input name. It takes in an optional
-//   share-io-dir argument to share an IO directory with an existing server.
-//   This is useful when restoring from a backup taken in another server.
+// - "new-server name=<name> [args]"
+//   Create a new server with the input name.
+//
+//   Supported arguments:
+//
+//   + share-io-dir: can be specified to share an IO directory with an existing
+//   server. This is useful when restoring from a backup taken in another
+//   server.
+//
+//   + allow-implicit-access: can be specified to set
+//   `EnableNonAdminImplicitAndArbitraryOutbound` to true
+//
+//   + disable-http: disables use of external HTTP endpoints.
+//
+//   + temp-cleanup-freq: specifies the frequency with which the temporaru table
+//   cleanup reconciliation job runs
 //
-// - "exec-sql server=<name>"
+//   + localities: specifies the localities that will be used when starting up
+//   the test cluster. The cluster will have len(localities) nodes, with each
+//   node assigned a locality config corresponding to the locality. Please
+//   update the `localityCfgs` map when adding new localities.
+//
+//   + nodes: specifies the number of nodes in the test cluster.
+//
+//   + splits: specifies the number of ranges the bank table is split into.
+//
+// - "exec-sql [server=<name>] [user=<name>] [args]"
 //   Executes the input SQL query on the target server. By default, server is
 //   the last created server.
 //
-//   Supported options:
+//   Supported arguments:
+//
+//   + expect-error-ignore: expects the query to return an error, but we will
+//   ignore it.
 //
-// - "query-sql server=<name>"
+// - "query-sql [server=<name>] [user=<name>]"
 //   Executes the input SQL query and print the results.
 //
 // - "reset"
 //    Clear all state associated with the test.
+//
+// - "job" [server=<name>] [user=<name>] [args]
```

Comment:
I don't yet see the reason for these new  job specific (e.g. backup, restore, schema change) data driven commands, if they all have the same args. Why not keep all common job args under the `job` cmd?

Reply:
> if they all have the same args. Why not keep all common job args under the job cmd?

good question! I started off with this, but the thing is that when tagging a job/expecting a pausepoint we actually need the type of the job to find the job ID - https://github.com/cockroachdb/cockroach/pull/78430/files#diff-28ef805944e40c647f9dbbba0d4b566a3eb4754e51d639e8e6693767bc7c0592R579. i.e. we pickup the latest job of a paritcular job type.

In my first iteration I had `job ... type='<job type string>` but this quickly became cumbersome cause the user had to enter the job type as it was persisted in the jobs table. Given that we should only see 4 ish jobs in this testing package import, backup, restore, and schema change I think having individual commands for each job type is a better UX. Furthermore, there are some properties like aost that only apply to a subset of the job types. If I had a shared command in the `job` instruction I would have to perform some custom validation to ensure `aost` is only supplied for `backup`, and `restore` etc.  By having a dedicated command for a specific job type we implicitly ensure that certain instruction arguments can only be provided to certain kinds of jobs.

I have pulled out all the shared logic into helpers though to ensure minimal code repetition.

Reply:
> also i think the aost arg is missing from the docs

nice catch, added